src/main/java/seedu/sellsavvy/logic/parser/AddOrderCommandParser.java

```diff
@@ -0,0 +1,65 @@
+package seedu.sellsavvy.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.sellsavvy.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_COUNT;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.sellsavvy.logic.parser.CliSyntax.PREFIX_ITEM;
+
+import java.util.stream.Stream;
+
+import seedu.sellsavvy.commons.core.index.Index;
+import seedu.sellsavvy.logic.commands.ordercommands.AddOrderCommand;
+import seedu.sellsavvy.logic.parser.exceptions.ParseException;
+import seedu.sellsavvy.model.order.Count;
+import seedu.sellsavvy.model.order.Date;
+import seedu.sellsavvy.model.order.Item;
+import seedu.sellsavvy.model.order.Order;
+
+/**
+ * Parses input arguments and creates a new AddOrderCommand object.
+ */
+public class AddOrderCommandParser implements Parser<AddOrderCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddOrderCommand
+     * and returns an AddOrderCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public AddOrderCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_ITEM, PREFIX_DATE, PREFIX_COUNT);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
```

Comment:
maybe you can use !argMultimap.getPreamble().isEmpty() to check if preamble is empty first.

Reply:
like combine checking for preamble together with line 44 i.e move line 44 up

Reply:
`!argMultimap.getPreamble().isEmpty()` returns true the preamble is **not** empty and is used for commands where you're not supposed to put anything as the preamble.
For this command you must fill in the index into the preamble. `ParserUtil.parseIndex` also throws a `ParseException` if the preamble is empty.

Reply:
I feel that we should check it first and not use try - catch as a way to detect wrong input. Basically not using exception as control flow mechanism like what we learnt in cs2030S.
![image](https://github.com/user-attachments/assets/0127532f-37d4-4f69-83d3-67bd93cf5399)
anyway i meant argMultimap.getPreamble().isEmpty() no !.

Reply:
ok sure, will change